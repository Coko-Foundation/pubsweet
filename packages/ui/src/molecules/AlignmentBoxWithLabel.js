import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import fromTheme from '../helpers/fromTheme'

import AlignmentBox from '../atoms/AlignmentBox'

const blue = '#0d78f2'

const Root = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
`

const Label = styled.span`
  color: ${blue};
  font-family: ${fromTheme('fontInterface')};
  font-size: 16px;
  font-style: italic;
  margin-left: ${props => (props.positionRight ? '10px' : '0')};
  margin-right: ${props => (props.positionRight ? '0' : '10px')};
  order: ${props => (props.positionRight ? '2' : '0')};
`

const AlignmentBoxWithLabel = ({
  active,
  id,
  labelPositionRight,
  labelText,
  noBorder,
  onClick,
}) => (
  <Root>
    <Label positionRight={labelPositionRight}>{labelText}</Label>
    <AlignmentBox
      active={active}
      id={id}
      noBorder={noBorder}
      onClick={onClick}
    />
  </Root>
)

AlignmentBoxWithLabel.propTypes = {
  active: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  labelPositionRight: PropTypes.bool,
  labelText: PropTypes.string.isRequired,
  noBorder: PropTypes.shape({
    top: PropTypes.bool,
    bottom: PropTypes.bool,
    right: PropTypes.bool,
    left: PropTypes.bool,
  }),
  onClick: PropTypes.func,
}

AlignmentBoxWithLabel.defaultProps = {
  labelPositionRight: false,
  noBorder: {
    top: false,
    bottom: false,
    right: false,
    left: false,
  },
  onClick: () => null,
}

export default AlignmentBoxWithLabel
