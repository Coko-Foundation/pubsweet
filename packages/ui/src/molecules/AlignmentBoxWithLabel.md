AligmentBox with Label on the left

```js

const active = true;
const id = 'eyedee';
const labelText = 'left';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBoxWithLabel
  active={active}
  id={id}
  labelText={labelText}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Inactive AligmentBox with Label on the right

```js

const active = false;
const id = 'eyedee';
const labelPositionRight = true;
const labelText = 'right';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBoxWithLabel
  active={active}
  id={id}
  labelPositionRight={labelPositionRight}
  labelText={labelText}
  noBorder={noBorder}
  onClick={onClick}
/>
```
