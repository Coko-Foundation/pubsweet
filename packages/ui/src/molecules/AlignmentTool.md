AligmentTool with left box active

```js
const data = [
  {
    active: true,
    id: 'left',
    label: 'left'
  },
  {
    active: false,
    id: 'right',
    label: 'right'
  }
];

const onClickAlignmentBox = () => null;

<AlignmentTool
  data={data}
  onClickAlignmentBox={onClickAlignmentBox}
/>
```
