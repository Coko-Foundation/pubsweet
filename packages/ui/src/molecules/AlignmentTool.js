import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { AlignmentBoxWithLabel } from '../molecules'

const mainGrey = '#666'

const Root = styled.div`
  display: flex;
  margin-left: auto;
`

const MiddleLine = styled.div`
  background-color: ${mainGrey};
  height: 40px;
  width: 1px;
`

const AlignmentTool = ({ data, onClickAlignmentBox }) => {
  const onClick = event => {
    const { id } = event.currentTarget
    onClickAlignmentBox(id)
  }

  const leftData = data[0]
  const rightData = data[1]

  const noBorderRight = { right: true }
  const noBorderLeft = { left: true }

  return (
    <Root>
      <AlignmentBoxWithLabel
        active={leftData.active}
        id={leftData.id}
        labelText={leftData.label}
        noBorder={noBorderRight}
        onClick={onClick}
      />

      <MiddleLine />

      <AlignmentBoxWithLabel
        active={rightData.active}
        id={rightData.id}
        labelPositionRight
        labelText={rightData.label}
        noBorder={noBorderLeft}
        onClick={onClick}
      />
    </Root>
  )
}

AlignmentTool.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool.isRequired,
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    }),
  ).isRequired,
  onClickAlignmentBox: PropTypes.func.isRequired,
}

export default AlignmentTool
