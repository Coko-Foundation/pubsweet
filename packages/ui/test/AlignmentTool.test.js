import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import AlignmentBoxWithLabel from '../src/molecules/AlignmentBoxWithLabel'
import AlignmentTool from '../src/molecules/AlignmentTool'

const requiredProps = {
  data: [
    {
      active: false,
      id: 'left',
      label: 'Left',
    },
    {
      active: false,
      id: 'right',
      label: 'right',
    },
  ],
  onClickAlignmentBox: () => null,
}

const wrapper = shallow(<AlignmentTool {...requiredProps} />)
const wrapperMounted = mount(<AlignmentTool {...requiredProps} />)
const alignmentBoxesWithLabel = wrapper.find(AlignmentBoxWithLabel)

describe('AlignmentTool', () => {
  test('is rendered correctly', () => {
    const tree = renderer.create(<AlignmentTool {...requiredProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should contain two Alignment Boxes with label', () => {
    expect(alignmentBoxesWithLabel.exists()).toEqual(true)
    expect(alignmentBoxesWithLabel).toHaveLength(2)
  })

  test('gets the correct props', () => {
    expect(wrapperMounted.prop('data')).toEqual(requiredProps.data)
    expect(wrapperMounted.prop('onClickAlignmentBox')).toEqual(
      requiredProps.onClickAlignmentBox,
    )
  })
})
