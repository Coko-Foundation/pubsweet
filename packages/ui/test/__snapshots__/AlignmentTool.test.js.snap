// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentTool is rendered correctly 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Left
    </span>
    <div
      className="sc-bdVaJa ieYRgr"
      id="left"
      onClick={[Function]}
      role="presentation"
    />
  </div>
  <div
    className="middleLine"
  />
  <div
    className="root reverseOrder"
  >
    <span
      className="label"
    >
      right
    </span>
    <div
      className="sc-bdVaJa fCBdDA"
      id="right"
      onClick={[Function]}
      role="presentation"
    />
  </div>
</div>
`;
